### PR TITLE
Reissue Certificate Emails

### PR DESCRIPTION
**Pull request by bbengfort**
### Scope of changes

This PR creates three new email templates as well as the context data and helper functions to send them. The email templates are:

1. "ExpiresAdminNotification": the email sent to the TRISA admins when a certificate is about to expire so they can review the certs and ensure that they should be reissued.
2. "ReissueReminder": the email sent to all verified contacts associated with the VASP letting them know that their certificates will expire and be automatically reissued soon.
3. "ReissueStarted": the email sent to the same contact that will receive the certs (only one contact -- usually the technical contact) that contains the whisper link to the PKCS12 password. 

Each of these emails has:

1. Templates (both text and html) in the templates directory 
2. A subject for the email
3. A context `Data` structure to populate the templates 
4. An email builder function to create the email from the templates and the contex
5. A client send method to quickly send the email using send grid

### Type of change

- [ ] bug fix
- [x] new feature
- [ ] documentation
- [ ] other (describe)

### Acceptance criteria

Grammar and consistency are important for these emails. Also please check that we're sending to the right contacts and including the right information in the email template! 

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [x]  I have recompiled and included new protocol buffers to reflect changes I made
- [x]  I have added new test fixtures as needed to support added tests
- [ ]   Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [x] QA `expires_admin_notification` email for grammar and spelling (including subject)
- [x] QA `reissuance_remdiner` email for grammar and spelling (including subject)
- [x] QA `reissuance_started` email for grammar and spelling (including subject)
- [ ] Tests seem consistent 
- [x] Data in contexts is appropriate 
- [x] Emails make sense generally

### CODE REVIEW

File: pkg/gds/emails/client.go [hunk 1]
```diff
@@ -325,3 +328,160 @@ func (m *EmailManager) SendDeliverCertificates(vasp *pb.VASP, path string) (sent
...
+		return 0, err
+	}
+
+	return 1, nil
+}
+
+// SendReissuanceReminder sends a reminder to all verified contacts that their identity
+// certificates will be expiring soon and that the system will automatically reissue the
+// certs on a particular date.
+func (m *EmailManager) SendReissuanceReminder(vasp *pb.VASP, reissueDate time.Time) (sent int, err error) {
```
Comment (bbengfort) [inline pkg/gds/emails/client.go:371]: If we can get this merged in by Wednesday, sending the reminders may be as simple as pressing a button :-)

**Review by rebeccabilbro — APPROVED** [review]
This looks good to me @bbengfort! I'll defer to @nwhite222 on the wording of the emails (the grammar seems ok at least). The code and tests all look great and I'm glad those emails we drafted a few weeks ago came in handy here 🙂 . Also good thinking in updating the support email to one that's sure to always get a response 😉

File: pkg/gds/emails/templates/reissuance_reminder.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+Hello {{ .Name }},
+
+We wish to remind you that your TRISA Identity Certificates for {{ .CommonName }} will soon be expiring. The expiration date on your current certificates is {{ .ExpirationDate }}.
+
+The Global Directory Service will automatically reissue your certificates on {{ .ReissueDate }}. When the certificates are reissued, you will receive two additional emails; one of which will contain your password-protected certificates and the other of which will contain your PKCS12 password, which you will use to decrypt the new certificates.
```
Comment (nwhite222) [inline pkg/gds/emails/templates/reissuance_reminder.txt:5]: I would re-work the second and third sentences this way (taking out the semicolon): 
"When the certificates are reissued, you will receive two additional emails - one will contain your password-protected certificates and the other will contain your PKCS12 password. You will use the PKCS12 password to decrypt the new certificates."

**Review by nwhite222 — COMMENTED** [review]
Looks great! I made one small grammatical suggestion on the 'Reissuance Reminder' template, but otherwise, nothing to add.

**Comment by bbengfort** [comment]
@rebeccabilbro the Admin UI tests are failing but I'm going to go ahead and merge; it doesn't appear to be anything related to this PR.

**Pull request merged**